components/segmenter/src/complex/lstm/mod.rs

```diff
@@ -71,19 +71,32 @@ impl<'l> LstmSegmenter<'l> {
     /// Returns `Err` if grapheme data is required but not present
     pub(super) fn new(lstm: &'l LstmDataV1<'l>, grapheme: &'l RuleBreakDataV1<'l>) -> Self {
         let LstmDataV1::Float32(lstm) = lstm;
+        let fw_w = MatrixZero::from(&lstm.fw_w);
+        let fw_u = MatrixZero::from(&lstm.fw_u);
+        let bw_w = MatrixZero::from(&lstm.bw_w);
+        let bw_u = MatrixZero::from(&lstm.bw_u);
         let time_w = MatrixZero::from(&lstm.time_w);
+
+        let hunits = fw_w.dim().1;
+        let embedd_dim = fw_w.dim().2;
+
+        let fw_w = fw_w.reshape([4 * hunits, embedd_dim]);
```

Comment:
suggestion: instead of a general `reshape`, define a `collapse_1_2` to collapse the first two dimensions. Then you don't have to compute the sizes here.